test-data/unit/merge.test

```diff
@@ -669,17 +669,17 @@ TypeInfo<2>(
   Mro(target.N<2>, builtins.tuple<3>, typing.Sequence<4>, typing.Iterable<5>, builtins.object<1>)
   Names(
     _NT<6>
-    __annotations__<7> (builtins.object<1>)
-    __doc__<8> (builtins.str<9>)
-    __new__<10>
-    _asdict<11>
-    _field_defaults<12> (builtins.object<1>)
-    _field_types<13> (builtins.object<1>)
-    _fields<14> (Tuple[builtins.str<9>])
-    _make<15>
-    _replace<16>
-    _source<17> (builtins.str<9>)
-    x<18> (target.A<0>)))
+    __annotations__<7> (builtins.dict[builtins.str<8>, Any]<9>)
+    __doc__<10> (builtins.str<8>)
+    __new__<11>
+    _asdict<12>
+    _field_defaults<13> (builtins.dict[builtins.str<8>, Any]<9>)
+    _field_types<14> (builtins.dict[builtins.str<8>, Any]<9>)
+    _fields<15> (Tuple[builtins.str<8>])
+    _make<16>
+    _replace<17>
+    _source<18> (builtins.str<8>)
+    x<19> (target.A<0>)))
```

Comment:
Would be great if someone could have a look at this and verify that it's correct.